design/src/rfcs/rfc0005_service_generation.md

```diff
@@ -0,0 +1,76 @@
+RFC: Smithy Rust Service Framework
+==================================
+
+> Status: RFC
+
+The Rust Smithy Framework is a full-fledged service framework whose main
+responsibility is to handle request lifecycles from beginning to end. It takes
+care of input de-serialization, operation execution, output serialization,
+error handling, and provides facilities to fulfill the requirements below.
+
+Requirements
+------------
+
+### Smithy model-driven code generation
+
+Server side code is generated from Smithy models and implements operations,
+input and output structures, and errors defined in the service model.
+
+### Performance
+
+This new framework is built with performance in mind. It refrains from
+allocating memory when not needed and try to use a majority of
+[borrowed](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) types,
+handling their memory lifetimes so that a request body can be stored in memory
+only once and not
```

Comment:
If you wanted to use a $10 word you could say that we're parsimonious about memory